libs/events/events.go

```diff
@@ -194,14 +203,15 @@ func newEventListener(id string) *eventListener {
 	}
 }
 
-func (evl *eventListener) AddEvent(event string) {
+func (evl *eventListener) AddEvent(event string) error {
 	evl.mtx.Lock()
 	defer evl.mtx.Unlock()
 
 	if evl.removed {
-		return
+		return fmt.Errorf("listener was removed.")
```

Comment:
`var ErrListenerRemoved = errors.New("listener was removed")`

Reply:
that is ok， I'll replace other error message with named error too.